### PR TITLE
Add table query feature

### PR DESCRIPTION
**Pull request by zzhxiaofeng**
- Query table implementation & tests
- Added functional test cases to table.test.ts
#484

### CODE REVIEW

File: tests/table/apis/table.test.ts [hunk 1]
```diff
@@ -126,6 +126,112 @@ describe("table APIs test", () => {
...
+    - application/json;odata=fullmetadata
+    */
+    requestOverride.headers = {
+      accept: "application/json;odata=fullmetadata"
+    };
+
+    tableService.queryEntities(
+      "Tables",
+      null,
+      null,
```
Comment (blueww) [inline tests/table/apis/table.test.ts:143]: Please refer https://docs.microsoft.com/en-us/rest/api/storageservices/query-entities, and make sure cover different filter(or pk,rk), and select in the test cases.

Comment (zzhxiaofeng) [reply tests/table/apis/table.test.ts:143]: Sorry, This is query table unit test and I use the wrong function.

File: src/table/handlers/TableHandler.ts [hunk 2]
```diff
@@ -116,25 +116,72 @@ export default class TableHandler extends BaseHandler implements ITableHandler {
...
+      date: context.startTime,
+      statusCode: 200,
+      xMsContinuationNextTableName: options.nextTableName,
+      value: []
+    };
+
+    let protocol = "http";
+    let host =
+      DEFAULT_TABLE_SERVER_HOST_NAME + ":" + DEFAULT_TABLE_LISTENING_PORT;
+    // TODO: Get host and port from Azurite Server instance
```
[the remark this answers is not in the document]
Comment (zzhxiaofeng) [reply src/table/handlers/TableHandler.ts:152]: I have created https://github.com/Azure/Azurite/issues/539 to track this.

File: src/table/handlers/TableHandler.ts [hunk 3]
```diff
@@ -116,25 +116,72 @@ export default class TableHandler extends BaseHandler implements ITableHandler {
...
+    ) {
+      throw StorageErrorFactory.getContentTypeNotSupported(context);
+    }
+
+    if (accountName === undefined) {
+      throw StorageErrorFactory.getAccountNameEmpty(context);
+    }
+
+    const metadata = `${accountName}/$metadata#Tables`;
+    const tableResult = await this.metadataStore.queryTable(context);
```
Comment (XiaoningLiu) [inline src/table/handlers/TableHandler.ts:137]: Hi @zzhxiaofeng, can you add `accountName` into `queryTable` parameter list? We want to pirortise critial parameters in metadata layer method as required parameters.

File: src/table/handlers/TableHandler.ts [hunk 4]
```diff
@@ -116,25 +116,72 @@ export default class TableHandler extends BaseHandler implements ITableHandler {
...
+      });
+    }
+
+    if (tableCtx.accept === FULL_METADATA_ACCEPT) {
+      response.odatametadata = `${protocol}://${host}/${metadata}`;
+      response.value = tableResult.map(item => {
+        return {
+          odatatype: item.odatatype,
+          odataid: item.odataid,
+          odataeditLink: item.odataeditLink,
```
Comment (XiaoningLiu) [inline src/table/handlers/TableHandler.ts:177]: @isabella0428 Please help check if some properties/links needs customize port dynamically.

Comment (isabella0428) [reply src/table/handlers/TableHandler.ts:180]: Yes，these fields should be set dynamically. 
Azurite's setting like host and port can be modified and the deprecated fields like editLink cannot be used to modify the objects.

**Pull request merged**